kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -499,6 +502,59 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         KRATOS_CATCH("")
     }
 
+    /**
+     * @brief The same methods as the base class but with constraints
+     * @param pScheme The pointer to the integration scheme
+     * @param rModelPart The model part to compute
+     * @param rb The RHS vector of the system of equations
+     */
+    void BuildRHSWithConstraints(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemVectorType& rb
+        )
+    {
+        Timer::Start("Build RHS");
```

Comment:
I am a little lost here.

What i would do is something like:

BuildRHS:
1 -  given the values of the master unknowns in the database, compute the value of the slaves
2 - loop over elements and conditions and assemble the values of the residual on all the nodes

in the Build()
i would do something very similar:
1 -  given the values of the master unknowns in the database, compute the value of the slaves
2 - build LHS and RHS
3 - apply constraints on both LHS and RHS

after solve (for example in the BuildAndSolve? one needs to update the value of the constraints

Reply:
Well, this is basically what is done

Reply:
Remember my approach is building a global matrix, but this is basically what is done